src/extension/app/components/plugin/plugin.js

```diff
@@ -142,6 +142,14 @@ export class Plugin {
     return this.config.container;
   }
 
+  /**
+   *  Is this plugin a badge?
+   * @returns {boolean} True if the plugin elements array has exactly one element and that element is a badge, otherwise false.
+   */
+  isBadge() {
+    return this.config.elements?.length === 1 && this.config.elements[0].tag === 'badge';
```

Comment:
Why don't you just look for a `this.config.isBadge` property? This would eliminate the need to have an intermediate `badge` element.